src/App.tsx

```diff
@@ -15,48 +19,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getSortedGoods(
+  goods: string[],
+  sortParameter: SortType,
+  reverseParameter: ReverseType,
+) {
+  const sortedGoods = [...goods];
+
+  if (sortParameter) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortParameter) {
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseParameter === ReverseType.Reverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortParameter, setSortParameter] = useState(SortType.Default);
+  const [reverseParameter, setReverseParameter] = (
+    useState(ReverseType.Straight)
+  );
+  const visibleGoods = (
+    getSortedGoods(goodsFromServer, sortParameter, reverseParameter)
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortParameter(SortType.Alphabet)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortParameter !== SortType.Alphabet,
+          })}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortParameter(SortType.Length)}
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortParameter !== SortType.Length,
+          })}
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => {
+            if (reverseParameter === ReverseType.Straight) {
+              setReverseParameter(ReverseType.Reverse);
+            } else {
+              setReverseParameter(ReverseType.Straight);
+            }
+          }}
```

Comment:
```suggestion
          onClick={() => 
              setReverseParameter(reverseParameter === ReverseType.Straight 
                  ? ReverseType.Reverse 
                  : ReverseType.Straight
           )}
```

Reply:
Thanks! I tried to make ternary with the function, not with the argument.